Raven.Database/Config/Categories/ServerConfiguration.cs

```diff
@@ -8,6 +9,19 @@ namespace Raven.Database.Config.Categories
 {
     public class ServerConfiguration : ConfigurationCategory
     {
+        /// <summary>
+        /// This much time has to pass before 
+        /// </summary>
+        [DefaultValue(15)]
+        [TimeUnit(TimeUnit.Seconds)]
+        [ConfigurationEntry(Constants.ConcurrentResourceLoadTimeout)]
+        [ConfigurationEntry(Constants.ConcurrentResourceLoadTimeout + "InSec")]
+        public TimeSetting ConcurrentResourceLoadTimeout { get; set; }
+
+        [DefaultValue(8)]
+        [ConfigurationEntry(Constants.MaxConcurrentResourceLoads)]
+        public int MaxConcurrentResourceLoads { get; set; }
+
```

Comment:
Thanks. I will make the necessary refactorings.

On Mon, Nov 16, 2015 at 9:57 PM, Arkadiusz Paliński <
notifications@github.com> wrote:

> In Raven.Database/Config/Categories/ServerConfiguration.cs
> <https://github.com/ayende/ravendb/pull/2862#discussion_r44974111>:
>
> > @@ -8,6 +9,19 @@ namespace Raven.Database.Config.Categories
> >  {
> >      public class ServerConfiguration : ConfigurationCategory
> >      {
> > +        /// <summary>
> > +        /// This much time has to pass before
> > +        /// </summary>
> > +        [DefaultValue(15)]
> > +        [TimeUnit(TimeUnit.Seconds)]
> > +        [ConfigurationEntry(Constants.ConcurrentResourceLoadTimeout)]
> > +        [ConfigurationEntry(Constants.ConcurrentResourceLoadTimeout + "InSec")]
> > +        public TimeSetting ConcurrentResourceLoadTimeout { get; set; }
> > +
> > +        [DefaultValue(8)]
> > +        [ConfigurationEntry(Constants.MaxConcurrentResourceLoads)]
> > +        public int MaxConcurrentResourceLoads { get; set; }
> > +
>
> @myarichuk <https://github.com/myarichuk> Please consider my comments:
>
> 1) we are moving away from the constants usage when we declare settings..
> if we really need to get the key of a setting we have RavenConfiguration.GetKey(x
> => x.CategoryName.Setting), however the current configuration handling
> limits such usages
> 2) please leave just the "InSec" name.. other settings still have old
> names configured (might be useful for 4.0 documentation purposes to write
> about introduced changes), however they are going to be removed as well (in
> 4.0 won't be backward compatible anyway)
> 3) TenantsConfiguration seems to be a better matched class for those new
> settings:
>
> https://github.com/ayende/ravendb/blob/v4.0/Raven.Database/Config/Categories/TenantConfiguration.cs
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ayende/ravendb/pull/2862/files#r44974111>.
>



-- 
Best regards,



*Michael Yarichuk*

RavenDB Core Team

Tel: 972-4-6227811

Fax:972-153-4-6227811

Email : michael.yarichuk@hibernatingrhinos.com



*Raven**DB** paving the way to "Data Made Simple" http://ravendb.net/
<http://ravendb.net/>  *